packages/crypto/src/public_key.ts

```diff
@@ -55,8 +70,12 @@ export class PublicKey extends Assignable {
             throw new Error('Invalid encoded key format, must be <curve>:<encoded key>');
         }
         const decodedPublicKey = baseDecode(publicKey);
-        if(decodedPublicKey.length !== KeySize.SECRET_KEY) {
-            throw new Error(`Invalid public key size (${decodedPublicKey.length}), must be ${KeySize.SECRET_KEY}`);
+        if (!keyType) {
+            keyType = decodedPublicKey.length === KeySize.SECP256k1_PUBLIC_KEY ? KeyType.SECP256K1 : KeyType.ED25519;
+        }
+        const keySize = keyType === KeyType.ED25519 ? KeySize.ED25519_PUBLIC_KEY : KeySize.SECP256k1_PUBLIC_KEY;
+        if(parts.length === 2 && decodedPublicKey.length !== keySize || parts.length === 1 && decodedPublicKey.length !== KeySize.ED25519_PUBLIC_KEY && decodedPublicKey.length !== KeySize.SECP256k1_PUBLIC_KEY) {
```

Comment:
Pretty sure this can be reduced to
```ts
if (decodedPublicKey.length !== keySize) { ... }
```

If `parts.length === 1`, then `keyType` was already determined from the length of the value. If an invalid length value (i.e. anything other than 64 or 32) were passed, `keyType` would be assumed to be ed25519 and fail because the length was not 32. Did I get that right?

Reply:
Yep, you are right, I will fix it.